JamGame/Assets/Scripts/Level/Shop/Room/View.cs

```diff
@@ -11,24 +13,37 @@ namespace Level.Shop.Room
     [AddComponentMenu("Scripts/Level/Shop/Room/Level.Shop.Room.View")]
     public class View : MonoBehaviour
     {
+        [RequiredIn(PrefabKind.Variant | PrefabKind.InstanceInScene)]
         [Pickle(LookupType = ObjectProviderType.Assets)]
         public CoreModel CoreModelPrefab;
 
         public InternalUid Uid => CoreModelPrefab.Uid;
 
+        [Required]
         [SerializeField]
-        private TMP_Text costText;
+        private TMP_Text costLabel;
 
-        private Func<CoreModel, Result> roomBuying;
+        [Required]
+        [SerializeField]
+        private TMP_Text countLabel;
+
+        private Action<CoreModel> roomBuying;
 
         [ReadOnly]
         [SerializeField]
-        private CoreModel coreModel;
-        public CoreModel CoreModel => coreModel;
+        private List<CoreModel> coreModels = new();
+        public bool IsEmpty => coreModels.Count == 0;
+        public CoreModel CoreModel => coreModels.Last();
```

Comment:
Do you actually need `CoreModel` here, or it can be replaced by `Uid`?

Reply:
Yes, reason that it's reference on game Object too, i want to change parent of it.